src/common/hooks/useHomepageCompareSliderMap.ts

```diff
@@ -0,0 +1,25 @@
+/**
+ * Used for Compare filter on homepage.
+ * Updates the MUI slider value whenever the filter scope changes on homepage.
+ */
+
+import { useEffect, useState } from 'react';
+import {
+  homepageScopeValueMap,
+  HomepageLocationScope,
+} from 'common/utils/compare';
+
+export default function useHomepageCompareSliderMap(
+  homepageScope: HomepageLocationScope,
+) {
+  const defaultHomepageSliderValue = homepageScopeValueMap[homepageScope];
+  const [homepageSliderValue, setHomepageSliderValue] = useState(
+    defaultHomepageSliderValue,
+  );
+
+  useEffect(() => {
+    setHomepageSliderValue(homepageScopeValueMap[homepageScope]);
+  }, [homepageScope]);
```

Comment:
I think `setHomepageSliderValue` needs to be a dependency as well

Reply:
hm the linter doesnt complain when its omitted. just double checked [in the docs](https://reactjs.org/docs/hooks-reference.html) - ("React guarantees that setState function identity is stable and won’t change on re-renders. This is why it’s safe to omit from the useEffect or useCallback dependency list.")

Reply:
nice!